### DISCUSSION TITLE
Unable to compile with SIMD enabled, macos

### DISCUSSION THREAD

**Discussion by MAG-AdrianMeredith** [jrouwe/JoltPhysics.js#153] [General]
Hey Everyone,
I was interested in trying out simd but I get an exception, any ideas? Is it working yet?  This is on macOS

<img width="1092" alt="image" src="https://github.com/jrouwe/JoltPhysics.js/assets/100127317/32337d54-97a0-48bc-b070-1bbc4e48dcf3">

**Comment by MAG-AdrianMeredith** [comment]
using this command,  works fine without
`./build.sh Distribution -DENABLE_SIMD=ON`

**Reply by MAG-AdrianMeredith** [reply]
I'm guessing "unhandled unaligned load" is the error here

**Comment by PhoenixIllusion** [comment]
The issue looks like it is in wasm2js, similar to the issue we had on multi-threading when it broke attempting to convert atomic-i64 operations to ASM.js (although those atomic ops technically exist via BigInt typed arrays). The ASM.js build should likely not be enabled for SIMD flags, since SIMD is a wasm-bytecode-only feature. ASM.JS (and FE javascript in general) has no concept of v128 variables or any SIMD operations.

If you build targeting just `wasm-compat` or `wasm` it should avoid this issue.
Using the flag BUILD_WASM_COMPAT_ONLY can force a working build, but if you also want to enable the `wasm` variant, it currently requires a manual approach.
Line 26 of build.sh: 
```bash
cmake --build Build/$BUILD_TYPE -j`nproc` --target jolt-wasm
```
The flag ENABLE_MULTI_THREADING automatically disables the ASM.js build as well, building just wasm & wasm-compat, but also forces you to use COOP/COEP on your server

**Comment by jrouwe** [comment]
I submitted a change with an extra check that should fix this problem.
